pytorch_lightning/plugins/training_type/tpu_spawn.py

```diff
@@ -201,12 +201,17 @@ def save_spawn_weights(self, model: LightningModule) -> Optional[str]:
             model.trainer.save_checkpoint(path)
             return path
 
+    def reduce_decision(self, decision: bool) -> bool:
+        decision = torch.tensor(int(decision), device=self.device)
+        decision = self.reduce(decision, "sum")
+        decision = bool(decision == self.world_size)
+        return decision
+
     def reduce_early_stopping_decision(self, should_stop: bool) -> bool:
-        should_stop = torch.tensor(int(should_stop), device=self.lightning_module.device)
-        stop = xm.mesh_reduce('stop_signal', should_stop, sum)
-        rendezvous("pl.EarlyStoppingCallback.stop_distributed_training_check")
-        should_stop = int(stop.item()) == self.world_size
-        return should_stop
+        return self.reduce_decision(should_stop)
+
+    def reduce_model_checkpoint_decision(self, should_update_best_and_save: bool) -> bool:
+        return self.reduce_decision(should_update_best_and_save)
```

Comment:
This is already in the parent parent, can remove.

But why don't we need all the `mesh_reduce` happiness anymore?

Reply:
We need the logic to convert the boolean for decision reduction.